packages/dcos-integration-test/extra/test_dcos_log.py

```diff
@@ -0,0 +1,100 @@
+import requests
+
+RANGE_ENDPOINT = '/system/v1/logs/v1/range/'
+STREAM_ENDPOINT = '/system/v1/logs/v1/stream/'
+
+
+def validate_json_entry(entry):
+    required_top_level_fields = ['fields', 'cursor', 'monotonic_timestamp', 'realtime_timestamp']
+
+    assert isinstance(entry, dict), 'Entry must be dict'
+    assert entry, 'Entry cannot be empty'
+
+    for required_field in required_top_level_fields:
+        assert required_field in entry, 'Field {} must be in entry. Got {}'.format(required_field, entry)
+
+    assert entry['fields'], '`fields` cannot be empty dict. Got {}'.format(entry)
+
+
+def validate_sse_entry(entry, stream=False):
+    lines = entry.split('\n')
+    assert len(lines) >= 2, 'Expect an least 2 lines of entry. Got {}'.format(entry)
+
+    data_line = {
+        True: lines[1],
+        False: lines[0]
+    }.get(stream)
+
+    entry_json = json.loads(data_line.lstrip('data: '))
+    validate_json_entry(entry_json)
+
+    if stream:
+        cursor = lines[0].lstrip('id: ')
+        assert cursor == entry_json['cursor'], 'In streaming mode an entry must have id: Got {}'.format(entry)
+
+
+def test_log_text(cluster):
+    for node in cluster.masters + cluster.all_slaves:
+        response = cluster.get(path=RANGE_ENDPOINT + '?limit=10', node=node)
+        assert response.ok
+        assert response.headers.get('Content-Type') == 'text/plain'
+
+        # expect 10 lines
+        lines = filter(lambda x: x != '', response.content.decode().split('\n'))
+        assert len(list(lines)) == 10, 'Expect 10 log entries. Got {}'.format(len(lines))
+
+
+def test_log_json(cluster):
+    for node in cluster.masters + cluster.all_slaves:
+        response = cluster.get(path=RANGE_ENDPOINT + '?limit=1', node=node, headers={'Accept': 'application/json'})
+        assert response.ok
+        assert response.headers.get('Content-Type') == 'application/json'
+        validate_json_entry(response.json())
+
+
+def test_log_server_sent_events(cluster):
+    for node in cluster.masters + cluster.all_slaves:
+        response = cluster.get(path=RANGE_ENDPOINT + '?limit=1', node=node, headers={'Accept': 'text/event-stream'})
+        assert response.ok
+        assert response.headers.get('Content-Type') == 'text/event-stream'
+        assert response.headers.get('Cache-Control') == 'no-cache'
+
+        validate_sse_entry(response.content.decode())
+
+
+def test_stream(cluster):
+    for node in cluster.masters + cluster.all_slaves:
+        response = cluster.get(path=STREAM_ENDPOINT + '?skip_prev=4', node=node, stream=True, timeout=2,
+                               headers={'Accept': 'text/event-stream'})
+        assert response.ok
+
+        lines = []
+        try:
+            buffer = ''
+            for line in response.iter_lines():
+                if not line:
+                    lines.append(buffer)
+                    buffer = ''
+                buffer += line.decode() + '\n'
+        except requests.exceptions.ConnectionError:
```

Comment:
why on connection error is it okay to silently move on? Needs a comment explaining